apps/dialtone-documentation/docs/components/rich-text-editor.md

```diff
@@ -0,0 +1,201 @@
+---
+title: Rich text editor
+description: A rich text editor is a component that allows users to format text with a variety of styles. It enables also mixing text with custom components built as extensions, such as links and user mentions.
+status: ready
+storybook: https://dialtone.dialpad.com/vue/?path=/story/components-rich-text-editor--default
+---
+
+<code-well-header>
+  <example-rich-text-editor :modelValue="defaultValue" />
+</code-well-header>
+
+## Base Style
+
+The editor itself is without any styling and the intention is to wrap it with another component, such as Message Input, that provides the UI.
+
+<code-well-header>
+  <example-rich-text-editor :modelValue="defaultValue" />
+</code-well-header>
+
+<code-example-tabs
+vueCode='
+<dt-rich-text-editor
+  v-model="value"
+  :editable="true"
+  input-aria-label="this is a descriptive label"
+  :input-class="$attrs.inputClass"
+  :output-format="$attrs.outputFormat"
+  :auto-focus="false"
+  placeholder="Type here..."
+  :link="true"
+/>
+'
+/>
+
+## With links
+
+<code-well-header>
+  <example-rich-text-editor
+    modelValue="<p>The editor can autolink URLs: <a target='_blank' rel='noopener noreferrer nofollow' class='d-link d-wb-break-all' href='http://dialpad.com'>dialpad.com</a>, <a target='_blank' rel='noopener noreferrer nofollow' class='d-link d-wb-break-all' href='https://www.dialpad.com/about-us/'>https://www.dialpad.com/about-us/</a>, email addresses: <a target='_blank' rel='noopener noreferrer nofollow' class='d-link d-wb-break-all' href='mailto:noreply@dialpad.com'>noreply@dialpad.com</a></p>"
+  />
+</code-well-header>
+
+## Output format
+
+There are 3 defined output formats currently for the input text.
+
+### Text
+
+Provide prop `outputFormat: 'text'`
+
+Plain text format as output without taking into consideration any marks (italics, bold) / html tags/ codeblocks
+that may be included in the input text user provides.
+
+### Json
+
+Provide prop `outputFormat: 'json'`
+
+the input will be formated as a json document that can then be consumed by product.
+The type of the each node in the input will be present for example: `codeblock`, `image`
+along with other meta data pertaining to the text node.
+
+### HTML
+
+Provide prop `outputFormat: 'html'`
+
+the input will be formated as native html that can then be consumed by product.
+
+## Tiptap Plugins Used
```

Comment:
There are a LOT more TipTap extensions we use than just this now, but I think this is okay to just get to the equivalent of the storybook docs.